src/styles/_catalog.scss

```diff
@@ -1,14 +1,32 @@
 .catalog {
-  display: flex;
+  display: grid;
   flex-wrap: wrap;
+  justify-content: center;
+  grid-template-columns: repeat(4, 200px);
+  gap: 48px;
 
   max-width: 944px;
   margin: 0 auto;
 
+
+  @media (max-width: 1023px) {
+    grid-template-columns: repeat(3, 200px);
+  }
+
+  @media (max-width: 768px) {
+    grid-template-columns: repeat(2, 200px);
+  }
+
+  @media (max-width: 487px) {
+    grid-template-columns: 200px;
+  }
```

Comment:
You can make a loop to create all media queries. Just create a correctly mapped list to iterate.